### PR TITLE
Users/v avirao/queue availability

### PR DESCRIPTION
**Pull request by v-avirao**
Initial changes to add queue availability

### CODE REVIEW

File: src/Interfaces/IGetQueueAvailabilityOptionalParams.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+export default interface IGetQueueAvailabilityOptionalParams {
+  /**
+   * JWT token from the portal
+   */
+  authenticatedUserToken?: string;
+  /**
+   * Whether to get init context from browser.
+   */
+  getContext?: boolean;
```
Comment (xTEddie) [inline src/Interfaces/IGetQueueAvailabilityOptionalParams.ts:11]: Is `getContext` needed?

Comment (xTEddie) [inline src/Interfaces/IGetQueueAvailabilityOptionalParams.ts:11]: Is `getContext` needed?

File: src/Interfaces/IGetQueueAvailabilityOptionalParams.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  /**
+   * Custom init context.
+   */
+  initContext?: InitContext;
```
Comment (xTEddie) [inline src/Interfaces/IGetQueueAvailabilityOptionalParams.ts:15]: Is `initContext` needed?

File: src/Interfaces/IGetQueueAvailabilityOptionalParams.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import InitContext from "../Model/InitContext";
+
+export default interface IGetQueueAvailabilityOptionalParams {
+  /**
+   * JWT token from the portal
+   */
+  authenticatedUserToken?: string;
```
Comment (xTEddie) [inline src/Interfaces/IGetQueueAvailabilityOptionalParams.ts:7]: Does this endpoint has an auth'ed endpoint? If not, then the `OptionalParams` is not needed.

File: src/Interfaces/IGetQueueAvailabilityOptionalParams.ts (hunk 1 above)
Comment (v-avirao) [reply src/Interfaces/IGetQueueAvailabilityOptionalParams.ts:11]: Yes, It is required to update the engagement context information

File: src/Interfaces/IGetQueueAvailabilityOptionalParams.ts (hunk 2 above)
Comment (v-avirao) [reply src/Interfaces/IGetQueueAvailabilityOptionalParams.ts:15]: Yes, InitContext is required to pass on the data to execute rules from the context variables and the engagement contexts

File: src/Interfaces/IGetQueueAvailabilityOptionalParams.ts (hunk 3 above)
Comment (v-avirao) [reply src/Interfaces/IGetQueueAvailabilityOptionalParams.ts:7]: Yes, this will be an authenticated end point

**Review by xTEddie — APPROVED** [review]

**Pull request merged**